src/metabase/util/malli/registry.cljc

```diff
@@ -66,7 +86,7 @@
   (letfn [(make-explainer []
             (try
               #_{:clj-kondo/ignore [:discouraged-var]}
-              (let [validator* (mc/validator schema)
+              (let [validator* (validator schema)
```

Comment:
We were actually compiling a whole additional validator for each schema here instead of reusing the once in the cache. Oops! Just cut memory usage in half or something.

Reply:
true https://github.com/metabase/metabase/pull/61291

Reply:
so I think 

```
(fn schema-explainer [value]
  (when-not (validator* value)
    (explainer* value)))

;; could be
(when-let [explained (explainer* value)] explained)
```

Right? any reason not to do it that way? making validators is good because they are smaller but if you already have to make an explainer, I don't see the purpose of making validators too

Reply:
Hmm maybe it's slower tho, that's the one reason i can think of to leave it like it is.